core/vm/instructions.go

```diff
@@ -342,9 +342,9 @@ func opExtCodeSize(pc *uint64, interpreter *EVMInterpreter, scope *ScopeContext)
 	slot := scope.Stack.peek()
 	address := slot.Bytes20()
 	if witness := interpreter.evm.StateDB.Witness(); witness != nil {
-		witness.AddCode(interpreter.evm.StateDB.GetCode(address))
+		witness.AddCode(interpreter.evm.StateDB.ResolveCode(address))
 	}
-	slot.SetUint64(uint64(interpreter.evm.StateDB.GetCodeSize(slot.Bytes20())))
+	slot.SetUint64(uint64(len(interpreter.evm.StateDB.ResolveCode(slot.Bytes20()))))
```

Comment:
I don't love this change, but we need to do it for EOF as well. It means that extcodesize gets a lot heavier now

Reply:
I wonder if it isn't better to define a new `opExtCodeSizeEip7702`. Instead of making ops more complex, we simply make new ops. ..?